test/unit/builtins/string.spec.ts

```diff
@@ -367,3 +367,15 @@ test("string intersected method", () => {
         return ({ abc: () => "a" } as Vector).abc();
     `.expectToMatchJsResult();
 });
+
+// Issue #1218: https://github.com/TypeScriptToLua/TypeScriptToLua/issues/1218
+test.each(['"foo"', "undefined"])("prototype call on nullable string (%p)", value => {
+    util.testFunction`
+        function toUpper(str?: string) {
```

Comment:
`str?: string` is one way of triggering the bug in #1218 - Is it also worth having tests for the other potential triggers, `str: string | undefined` and `str: string | null`? I only see tests for `str?: string`.

Reply:
They should all resolve to the same type (kinda), but you are right, adding some more tests wouldn't hurt. I added some more.